esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -239,15 +239,22 @@ void WsWuInfo::readWorkunitComponentLogs(const char* outFile, unsigned maxLogRec
     else if (logFormat == LOGACCESS_LOGFORMAT_xml)
         writeStringToStream(*outIOS, "<lines>");
 
-    while (logReader->readLogEntries(logcontent.clear(), recsRead))
+    bool theresMore = false;
+    do
     {
-        if (logFormat == LOGACCESS_LOGFORMAT_json && totalRecsRead > 0 && recsRead > 0)
+        theresMore = logReader->readLogEntries(logcontent.clear(), recsRead);
+
+        if (recsRead > 0)
         {
-            writeCharToStream(*outIOS, ',');
+            if (logFormat == LOGACCESS_LOGFORMAT_json && totalRecsRead > 0)
+            {
+                writeCharToStream(*outIOS, ',');
+            }
+            totalRecsRead += recsRead;
+            writeStringToStream(*outIOS, logcontent.str());
         }
-        totalRecsRead += recsRead;
-        writeStringToStream(*outIOS, logcontent.str());
     }
+    while (theresMore == true);
```

Comment:
picky:  while (theresMore) is easier to read.
            moreToRead is possibly a better variable name especially because "there's" has an apostrophe.

Reply:
agreed in principle but will insist we do not wait for this change if we're in the process of tagging within the next day or so.